indices/shuffle_graph.py

```diff
@@ -0,0 +1,42 @@
+import argparse
+import pickle
+import os
+from copy import deepcopy
+
+import networkx as nx
+from tqdm import tqdm
+
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser()
+    parser.add_argument('graph_path', help='The path to the graph to shuffle')
+    parser.add_argument('out_dir', help='The directory to store shuffled graphs in')
+    parser.add_argument('--n_graphs', help='The number of shuffled graphs to create', default=100)
+    args = parser.parse_args()
+
+    graph_path = args.graph_path
+    file_base = os.path.basename(graph_path)
+    file_base = os.path.splitext(file_base)[0]
+
+
+    with open(graph_path, 'rb') as in_file:
+        original_network = pickle.load(in_file)
+
+    n_edges = len(original_network.edges)
+    for i in tqdm(range(args.n_graphs)):
+        graph_copy = deepcopy(original_network)
+        # Directed edge swap swaps three edges at a time
+        n_swap = n_edges // 3
+
+        shuffled_graph = nx.directed_edge_swap(graph_copy,
```

Comment:
Right, I'm planning on waiting for nx version 3 to release then just pinning it as the required version. If I end up publishing the code before then I'll probably pull out the edge swap and put it in `utils.py` to avoid confusion